ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -432,6 +444,30 @@ private[ember] object Parser {
         (EmptyBody.covary[F], (Some(buffer): Option[Array[Byte]]).pure[F]).pure[F]
       }
 
+    def parseUnknownBody[F[_]: Concurrent](
+        buffer: Array[Byte],
+        read: Read[F]
+    ): F[(EntityBody[F], Drain[F])] =
+      Ref[F].of(false).map { consumed =>
+        lazy val readAll: Pull[F, Byte, Unit] =
+          Pull.eval(read).flatMap {
+            case Some(c) => Pull.output(c) >> readAll
+            case None => Pull.eval(consumed.set(true)).void
+          }
+
+        val body =
+          Stream
+            .eval(consumed.get)
+            .ifM(
+              ifTrue = Stream.raiseError(BodyAlreadyConsumedError()),
+              ifFalse = Stream.chunk(Chunk.array(buffer)) ++ readAll.stream
+            )
+
```

Comment:
Perhaps this would also work, 

```suggestion
          Pull.eval(consumed.get).flatMap {
            case true =>  Pull.raiseError(BodyAlreadyConsumedError())
            case false => Pull.output(Chunk.array(buffer)) >> readAll
          }.stream
```

Reply:
They appear equivalent, but if expressing via `Pull` is preferable, I will change.

Reply:
This might be a few less allocations and is still legible.  Mild :+1: